File: 김현욱/5주차/15649.cpp
#include <iostream>
using namespace std;

int n, m;
bool visited[9] = {}; 
int arr[9] = {}; 

void recursive(int k){
	if(m == k){
		for(int i = 0; i < m; i++)
			cout << arr[i] << " ";
		cout << "\n";
		return;
	}
	
	for(int i = 1; i <= n; i++){
		if(!visited[i]){
			visited[i] = true;
			arr[k] = i;
            recursive(k+1);
            visited[i] = false;
        }
    }
}

int main() {
	cin >> n >> m;
	
	recursive(0);
	
	return 0;
}

File: 이상욱/5주차/15649.java
import java.io.*;
import java.util.*;

public class Boj15649 {
    public static void main(String[] args) throws IOException {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter output = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer split = new StringTokenizer(input.readLine());

        int n = Integer.parseInt(split.nextToken());
        int m = Integer.parseInt(split.nextToken());
        int[] arr = new int[m];
        boolean[] visited = new boolean[n + 1];
        backTracking(n, m, arr, visited, output);

        output.flush();
        output.close();
    }

    public static void backTracking(int n, int m, int[] arr, boolean[] visited, BufferedWriter output) throws IOException {
        if (m == 0) {
            for (int i = 0; i < arr.length; i++)
                output.write(arr[i] + " ");
            output.write("\n");
        }
        else {
            for (int i = 1; i <= n; i++) {
                if (visited[i])
                    continue;
                arr[arr.length - m] = i;
                visited[i] = true;
                backTracking(n, m - 1, arr, visited, output);
                visited[i] = false;
            }
        }
    }
}
